src/P5Functions/Math.js

```diff
@@ -426,13 +426,20 @@ module.exports = function withMath(p, undef) {
   * @see noise
   */
   p.random = function() {
-    if(arguments.length === 0) {
-      return internalRandomGenerator();
-    }
-    if(arguments.length === 1) {
-      return internalRandomGenerator() * arguments[0];
+    var aMin, aMax;
+    switch (arguments.length) {
+      case 0:
+        aMin = 0;
+        aMax = 1;
+        break;
+      case 1:
+        aMin = 0;
+        aMax = arguments[0];
+        break;
+      default:
+        aMin = arguments[0];
+        aMax = arguments[2];
```

Comment:
Are you sure you meant `arguments[2]` here? As second argument, this would be `argument[1]`. That said, with the rewrite above this would no longer be an issue